lib/modules/manager/poetry/artifacts.ts

```diff
@@ -187,7 +192,7 @@ export async function updateArtifacts({
       docker: {},
       toolConstraints: [{ toolName: 'python', constraint }],
       preCommands: [
-        `pip install --user ${quote(`poetry${poetryVersion ?? ''}`)}`,
+        `pip install --user ${quote(`poetry${poetryConstraint ?? ''}`)}`,
```

Comment:
can we convert this to tool install instead? containerbase is prepared.

- https://github.com/renovatebot/renovate/pull/19973

Reply:
Solve the same problem?

Reply:
yes

Reply:
ok, rewritten